rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m

```diff
@@ -0,0 +1,171 @@
+//  Created by Uladzislau Volchyk on 12/3/20.
+//
+
+#import "UVFeedItemWebViewController.h"
+#import <WebKit/WebKit.h>
+#import "UIBarButtonItem+PrettiInitializable.h"
+
+@interface UVFeedItemWebViewController () <WKNavigationDelegate>
+
+@property (nonatomic, retain) WKWebView *webView;
+
+@property (nonatomic, retain) UIBarButtonItem *goBackButton;
+@property (nonatomic, retain) UIBarButtonItem *goForwardButton;
+@property (nonatomic, retain) UIBarButtonItem *reloadWebPageButton;
+@property (nonatomic, retain) UIBarButtonItem *closeWebPageButton;
+@property (nonatomic, retain) UIBarButtonItem *openInBrowserButton;
+
+@end
+
+@implementation UVFeedItemWebViewController
+
+- (void)dealloc
+{
+    [_webView release];
+    [_goBackButton release];
+    [_goForwardButton release];
+    [_reloadWebPageButton release];
+    [_closeWebPageButton release];
+    [_openInBrowserButton release];
+    [super dealloc];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self setupLayout];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    self.navigationController.toolbarHidden = NO;
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+    [super viewWillDisappear:animated];
+    self.navigationController.toolbarHidden = YES;
+}
+
+// MARK: -
+
+- (void)setupLayout {
+    self.view.backgroundColor = UIColor.whiteColor;
+    [self.view addSubview:self.webView];
+    [NSLayoutConstraint activateConstraints:@[
+        [self.webView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+        [self.webView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
+        [self.webView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+        [self.webView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor]
+    ]];
+    
+    self.toolbarItems = @[
+        self.goBackButton,
+        [UIBarButtonItem fillerItem],
+        self.goForwardButton,
+        [UIBarButtonItem fillerItem],
+        self.reloadWebPageButton,
+        [UIBarButtonItem fillerItem],
+        self.closeWebPageButton,
+        [UIBarButtonItem fillerItem],
+        self.openInBrowserButton
+    ];
+}
+
+- (UIBarButtonItem *)webBarButtonItemWithImage:(UIImage *)image action:(SEL)selector {
+    return [[[UIBarButtonItem alloc] initWithImage:image
+                                             style:UIBarButtonItemStylePlain
+                                            target:self.webView
+                                            action:selector] autorelease];
+}
+
+// MARK: - Lazy
+
+- (WKWebView *)webView {
+    if(!_webView) {
+        _webView = [WKWebView new];
+        _webView.navigationDelegate = self;
+        _webView.translatesAutoresizingMaskIntoConstraints = NO;
+    }
+    return _webView;
+}
+
+- (UIBarButtonItem *)goBackButton {
+    if(!_goBackButton) {
+        _goBackButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-left"]
```

Comment:
Please make category for UIImage. 
NSString literal should be moved to global constant.

Reply:
nice point, done

Reply:
nice point, done

Reply:
Resolved.